### PR TITLE
Add auto publishing script.

### PR DESCRIPTION
**Pull request by amannn**
@sfussenegger I tried this out on one of our repos and it seems to work.

I tried adding this at the end:

```sh
# commit new version if it's a master build
if [ $branch = "master" ]; then
  git tag $newVersion
  git push origin --tags
fi
```

But got an error:

```
Permission denied (publickey).
fatal: Could not read from remote repository.
Please make sure you have the correct access rights
and the repository exists.
```

I left it out for now as it doesn't work, but I think this could be helpful. WDYT?

### CODE REVIEW

File: bin/publish.sh [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+major="$(echo $version | cut -d'.' -f1)"
+minor="$(echo $version | cut -d'.' -f2)"
+patch="$(echo $version | cut -d'.' -f3 | cut -d'-' -f1)"
+
+# increment patch version
+patch="$(($patch + 1))"
+newVersion="$major.$minor.$patch"
+
+# set prerelease suffix for non-master builds
+if [ $branch != "master" ]; then
```
Comment (sfussenegger) [inline bin/publish.sh]: I found that having default branch name configurable very useful for support of different versions. therefore we could make this configurable:

	function branchType () {
	  for b in ${DEFAULT_BRANCHES:-master}; do
	    [[ "$b" == "$1" ]] && echo default && return;
	  done
	  echo feature
	}

	# example usage
	case `branchType $branch` in
	"default")
	  echo default;
	  ;;
	*)
	  echo not default;
	  ;;
	esac

`DEFAULT_BRANCHES` could then be defined in pipelines settings to be `"master 1.x"` and builds from `master` and `1.x` would both be treated as release branches

**Review by sfussenegger — COMMENTED** [review]
hmm, pushing isn't supported out of the box. that's this issue https://bitbucket.org/site/master/issues/13213/push-back-to-remote-from-pipelines

**Comment by amannn** [comment]
Hmm, true. I'm watching that issue now so we can revisit this when there's an update.

File: bin/publish.sh (hunk 1 above)
Comment (amannn) [reply bin/publish.sh]: Good point. I've added this to the PR now. The code has been tested and seems to work.

**Review by sfussenegger — APPROVED** [review]

**Pull request merged**